addons/account/models/account_move.py

```diff
@@ -1826,10 +1826,14 @@ def write(self, vals):
 
         return res
 
-    def unlink(self):
+
+    @api.ondelete(at_uninstall=False)
+    def _move_posted_before(self):
         for move in self:
             if move.posted_before and not self._context.get('force_delete'):
                 raise UserError(_("You cannot delete an entry which has been posted once."))
```

Comment:
Could be updated to be like the one below (first check context then use an `any`)